File: src/main/java/com/example/kyrgyzstancentralmedicalcard/services/ConsentService.java
package com.example.kyrgyzstancentralmedicalcard.services;

import com.example.kyrgyzstancentralmedicalcard.entity.PatientConsent;

import java.util.List;

public interface ConsentService {
    List<PatientConsent> getByPatient(Long patientId);
    PatientConsent grant(Long patientId, PatientConsent consent);
    void revoke(Long patientId, Long consentId);
    boolean hasActiveConsent(Long patientId, Long organizationId);
}

File: frontend-react/src/services/consentService.js
import api from './api';

const getByPatientId = (patientId) => api.get(`/patients/${patientId}/consents`);
const grant = (patientId, data) => api.post(`/patients/${patientId}/consents`, data);
const revoke = (patientId, consentId) => api.delete(`/patients/${patientId}/consents/${consentId}`);

const consentService = { getByPatientId, grant, revoke };
export default consentService;
